pkg/capabilities/consensus/ocr3/capability.go

```diff
@@ -237,8 +239,13 @@ func (o *capability) queueRequestForProcessing(
 		WorkflowExecutionID: metadata.WorkflowExecutionID,
 		WorkflowID:          metadata.WorkflowID,
 		WorkflowOwner:       metadata.WorkflowOwner,
-		Observations:        i.Observations,
-		ExpiresAt:           o.clock.Now().Add(o.requestTimeout),
+		WorkflowName:        metadata.WorkflowName,
```

Comment:
the purpose of this is so the contract can whitelist the report name, right?

we need to document that wf author probably should not ever change the value of step ref name. a note here is probably good enough for now until we have real documentation

as i'm thinking about it, it would be better to have some decoupling between the graph implementation details and the contract API. a simple solution would be some sort of tagging that lets the dev express how a report maps to the contract whitelist.

it would make it much clearer, there would be less spooky-action-at-a-distance, and and it gives more flexibility to the interface b/c more than one report can be tagged with the same value, so the contract doesn't have to change if you add a new WF or new step.

is this tagging something that would have to be done before an audit?

note: i have first hand experience of production systems breaking precisely because of this workflow-step-by-reference pattern. it's a nightmare.

Reply:
@krehermann Resolving this as we're using a report id field in the config instead.